pyproject.toml

```diff
@@ -1,76 +1,78 @@
-[project]
+[tool.poetry]
 name = "ScopeSim"
-version = "0.7.1"
+version = "0.8.0a0"
 description = "Generalised telescope observation simulator"
-readme = "README.md"
-requires-python = ">=3.8"
-license = {text = "License :: OSI Approved :: GNU General Public License v3 (GPLv3)"}
-authors = [
-    {name = "Kieran Leschinski", email="kieran.leschinski@unive.ac.at"},
-]
+license = "GPL-3.0-or-later"
+authors = ["Kieran Leschinski <kieran.leschinski@unive.ac.at>"]
 maintainers = [
-    {name = "Kieran Leschinski", email="kieran.leschinski@unive.ac.at"},
-    {name = "Hugo Buddelmeijer", email="hugo@buddelmeijer.nl"},
+    "Kieran Leschinski <kieran.leschinski@unive.ac.at>",
+    "Hugo Buddelmeijer <hugo@buddelmeijer.nl>",
+    "Fabian Haberhauer <fabian.haberhauer@univie.ac.at>",
 ]
-classifiers=[
-    "Programming Language :: Python :: 3",
-    "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
-    "Operating System :: OS Independent",
+readme = "README.md"
+repository = "https://github.com/AstarVienna/ScopeSim"
+documentation = "https://scopesim.readthedocs.io/en/latest/"
+classifiers = [
     "Intended Audience :: Science/Research",
+    "Operating System :: OS Independent",
     "Topic :: Scientific/Engineering :: Astronomy",
 ]
-dependencies = [
-    "numpy>=1.20.0",
-    "scipy>=1.10.0",
-    "astropy>=5.0",
-    "matplotlib>=3.7.2",
-    "pooch>=1.7.0",  # for scipy.datasets
 
-    "docutils>=0.15",
-    "httpx>=0.23.0",
-    "beautifulsoup4>=4.4",
-    "lxml>=4.5.0",
-    "pyyaml>5.1",
-    "more-itertools>=9.0",
-    "tqdm>=4.66.1",
+[tool.poetry.dependencies]
+python = ">=3.8,<3.12"
```

Comment:
This `<3.12` makes our tests fail (https://github.com/AstarVienna/irdb/actions/runs/7297735519/job/19887475308?pr=150) because the testing matrix includes 3.12 (https://github.com/AstarVienna/DevOps/pull/8).

This problem was not caught because this PR `uses: AstarVienna/DevOps/.github/workflows/tests.yml@poetry`, where Python 3.12 is explicitly excluded (https://github.com/AstarVienna/DevOps/commit/1f9f5f19ecfd199e336c5abaa1602db215eeaa2c).

More importantly than our tests failing, is that we currently do not support Python 3.12 anymore! I'd rather have us drop support for Python 3.8 early, rather than adding Python 3.12 support late.

Ultimately it will be impossible to find versions of dependencies that work for all supported Python versions (see commit message in https://github.com/AstarVienna/DevOps/commit/1f9f5f19ecfd199e336c5abaa1602db215eeaa2c), so this current workflow is untenable.

Perhaps we can split the tests up: a single test using the poetry.lock file, and a full matrix test allowing free dependencies from pyproject.toml ?